### PR TITLE
Claro with CKEditor4 changes

### LINKED ISSUE #3537: Update CMS UI Modules
# Update CMS UI Modules

### Modules to be updated:
- admin_toolbar - upgrade from 3.0.3 -> 3.4.1
- moderation_sidebar - upgrade from 1.4.0-> 1.7.0 (D10 version)
-  1.7 Latest module version

### LINKED ISSUE #3545: Update linkit module
#Update linkit module

### Prerequisites
- linkit -> upgrade from 5.0.0-beta13 -> 6.0.0-rc1

### LINKED ISSUE #3829: Enabler: cgov_admin theme must change to use claro base theme
# cgov_admin theme must change to use claro base theme
> **ESTIMATE** TBD

The Drupal admin them we all know and love, seven, is replaced with the claro theme in D10. In order to be able to upgrade to Drupal 10, we must change our cgov_admin theme to use the new claro theme. The theme changes lots of colors, adds whitespace, and does some other funky things, but for the most part, everything stays in the same place. So the biggest part of this story may be updating training materials.

### Changes
* Content landing page
  * Breadcrumb moves to top of content window
  * Filter button is in line with filters (Title, Content type, Published Status, Language)
* Edit content items
  * If multiple paragraphs added, each paragraph displays collapsed in an accordion
    * NOTE: CTHP Cards are not in accordion and therefore all display expanded
  * Each item displays paragraph name, field values on paragraph (until no more characters can be displayed and it truncates), Edit button, and dropdown to Remove or Duplicate
  * Clicking Edit expands the accordion section for the paragraph to view and edit fields
  * Clicking Collapse closes the accordion section for the paragraph
  * Clicking Remove immediately deletes the paragraph (no Confirm Removal button)
  * Clicking Duplicate creates a duplicate paragraph with a caution triangle next to the Heading (???)
  * Show Row Weights displays at the top right of the accordion
  * Clicking Show Row Weights displays row weights next to each accordion section (starts at 0)
  * Clicking Hide Row Weights removes them from display next to each accordion section

### Acceptance Criteria
**Visiting the Content Page**
```gherkin
Scenario: A user visits and interacts with the Content page on Drupal
  Given a user goes to an active ODE site
    And the user logs in as an editor
    And the user goes to the Content page
  Then the user shall see that the breadcrumbs appear at the top left of the content window
  When the user clicks on any of the breadcrumbs
  Then the user will be redirected to the respective page as indicated by the breadcrumb
  When the user returns to the Content page
  Then they will see the Filter button is on the same horizontal line as the Filter options (Title, Content Type, Published Status, and Language)
```

**Editing Content**
```gherkin
Scenario: A content editor edits existing content on Drupal
  Given a user goes to an active ODE site
    And the user logs in as an editor
    And the user goes to the Content Page
    And the user selects the "Edit" button under Operations in the main table of available content
    And the user is redirected to the page of the "Edit <Content Type> <Title of Content Page>" page
  Then the users will see a language dropdown selection
    And the Contents section will contain the paragraphs
  When the Contents section contains multiple paragraphs
  Then the paragraphs will appear collapsed in an accordion
    And the collapsed paragraphs will display the paragraph name, field values on paragraph (until no more characters can be displayed and it truncates(, an Edit button, and a dropdown to Remove or Duplicate
  When the user clicks on the Edit button on a collapsed paragraph
  Then the paragraph will expand
    And the fields can be edited
  When the user clicks on the Collapse button that replaced the Edit button on expansion
  Then the paragraph will return to the collapsed state
  When the user is viewing a CTHP Cards
  Then they will not be included in the collapsed paragraphs section
    And they will appear in the expanded state
  When a user clicks the Remove button on a paragraph
  Then the item will immediately be deleted
  When the user clicks the Duplicate button on a paragraph
  Then an identical paragraph will appear will appear in the expanded state
  When a user clicks Collapse on the Duplicate paragraph
  Then a yellow caution icon will appear to the right of the paragraph title
    And on hover the icon will indicate to the user that they have unsaved changes on the paragraph
  When a user goes to the top of the Contents section
  Then the Show row weights button will appear at the top right of the section
  When the user clicks on the Show row weights button
  Then a column will be added to the paragraphs shown with the heading of "Order"
    And each paragraph will have a dropdown selection under "Order" starting with an index of 0
    And each subsequent paragraph will have a sequentially numbered index increasing by one from the previous paragraph
    And on selection of the dropdown, the user will be able to change the order index value of the paragraph
    And the four-directional arrow that allows the user to rearrange the order of the paragraphs on the left side of the collapsed paragraph section will disappear
  When the user clicks on the Hide row weights button at the top right of the Contents section, that replaced the Show row weights button
  Then the paragraphs will revert to their initial state that was shown prior to clicking Show row weights
  When a user views all other components of the page
  Then the names and functionality of the fields and components on the page will match the Drupal site prior to the theme change with changes made to the style of those other fields and components 
```

### Dependencies:
* #3484 - The newest version of paragraphs includes [3175491](https://www.drupal.org/project/paragraphs_asymmetric_translation_widgets/issues/3175491) which is needed for the Claro theme.

### Resources:
* Original notice: https://www.drupal.org/node/3087157

### Sub-Tasks
- [ ] Change `cgov_admin.info.yml` to use `claro`
- [ ] Make sure that existing sites do no break

### LINKED ISSUE #3950: Update token_filter, Entity Embed and Embed modules
I have updated to latest versions which are Drupal 10 ready but not CKeditor 5 yet, Development for those are still in progress and I found patches which are working fine in my local but not sure if those affect anything else. we have to look out for the stable patches for those to work without any issues.

CKeditor5 supports SVG images only, so we have to update all the entity embed icon images to SVG. There are some CSS issues in the editor for all the entity embed plugins which needs to checked after doing CKeditor5 CSS changes we mentioned in #3846 

Current version: “drupal/embed": "^1.6"
Latest version:   "drupal/embed": "1.x-dev@dev"

Current version: "drupal/entity_embed": "^1.2.0"
Latest version:   "drupal/entity_embed": "^1.4"

Current version: "drupal/token_filter":"^1.3.0"
Latest version: "drupal/token_filter":"^2.0"

Issues:
https://www.drupal.org/project/entity_embed/issues/3272732
https://www.drupal.org/project/embed/issues/3309747

NOTE: This is not production ready yet as there are patches still being worked on.

### LINKED ISSUE #4001: Update Paragraph to use paragraphs stable widget
Update and Paragraph config changes 

Related ticket: #3847 and #3986 

### Requirements
- Each field where paragraphs can be added has the "Add <Default Paragraph>" button
  - Example: "Add Primary Feature Card Row" to Contents field of Home and Landing content type
- Clicking on the `Add <Default Paragraph>` button adds to the field
- If multiple options, there is a down arrow that displays next to the `Add <Default Paragraph` option
- Clicking down arrow displays all paragraph options for that field
- Selecting an option from the list adds that paragraph to the field
- `Add <Default Paragraph>` button disappears if the max cardinality for the field has been reached
- Clicking "Remove" deletes the paragraph

### LINKED ISSUE #4085: CK Editor issue - Unable to see the Video Carousel in the CMS
When I am adding the Video Carousel (Display= Full, Alignment = Center), I am not able to see the video Carousel on the CMS.
Compare with the screen-shot on cms-dev (3rd screen-shot).

Image

Image

Image

### LINKED ISSUE #4099: Resolved with CK-Editor-4- Intro Text and Body section Alignment issue on the CMS
When we just click the "Edit" button and the content is loaded in edit mode, the Body section (Heading and Content) field and the Intro Text field are not aligned to each other. Tools does not wrap to the next line or does not appear collapsed under the drop-down (three dots) in the Body section (Content) field. Once we enter contents (type "test") to the ck-editor or click inside the ck-editor and then scroll down and then up, the Body section (Heading and Content) field and the Intro Text field gets aligned to each other. Is it ok?

Steps to reproduce:

1. Click Contents -> Add Contents-> Article
2. Scroll down to see that the Intro text WYSIWYG and Body Section is not aligned.
3. Notice that all the buttons in the tool bar (including the embedding buttons/icons) are visible.
4. Scroll to the bottom of the screen and then scroll up to the Body Section area
5. Still the Intro text WYSIWYG and Body Section is not aligned and all the buttons in the tool bar (including the embedding buttons/icons) are visible.
6. Now, just click inside the "Contents" WYSIWYG
7. Scroll all the way down
8. Scroll up and at one point you will see that the tool bar of the CK Editor rises up the menu bar (Screen-shot#2)   (https://github.com/orgs/NCIOCPL/projects/52/views/1?pane=issue&itemId=41369207)
9. Scroll up to the "Contents" WYSIWYG area under the Body Section
10. Notice that the the buttons (image, infographic, video, Insert Block Content, Insert Featured content, Glossify, Source) appear under the drop-down (that appears as three-dots) in the tool bar.

Image

Image

Image

### LINKED ISSUE #4122: Enabler: Replace EOL CKEditor4 with the LTS CKEditor.
# Replace EOL CKEditor4 with the LTS CKEditor.

CKEditor5 is not presently in a state to support our requirements. We need to switch to the [external ckeditor module](https://www.drupal.org/project/ckeditor) (as opposed to the EOL one in Drupal core).

### LINKED ISSUE #4123: Update platform to use CKEditor4 LTS Contributed module
Update the platform to use https://www.drupal.org/project/ckeditor_lts instead of the core ckeditor. This editor requires a license key so we will handle that via settings.

### LINKED ISSUE #4134: Spanish Translation - Warning message appears between the Intro text and Header when we edit the Body Section
On the Spanish Translation warning message appears between the Intro text and Header when we edit the Body Section. Earlier we used to see only 1 warning message at the top of the page.

Image

Image

This is known issue which got fixed in Drupal core 11.x-dev. Applying patch which is available for version 9.5.x

Issue link: https://www.drupal.org/project/drupal/issues/3002571

### PR DESCRIPTION
**Pull request by navsunka**
ODE: https://ncigovcdode647.prod.acquia-sites.com/

Closes #4001
Closes #3829
Closes #3537
Closes #3545
Closes #3950
Closes #4085
Closes #4099
Closes #4123
Closes #4122
Closes #4134

### CODE REVIEW

**Comment by KateMashkinaNIH** [comment]
Automated Regression tests passed

**Comment by alinai** [comment]
Manual Tests passed on ODE647 cc: @navsunka , @KateMashkinaNIH , @sarinapadilla , @santwuma

File: composer.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
         "drupal/ctools": "^4.0.0",
         "drupal/devel": "^5.1.2",
         "drupal/embed": "^1.7",
-        "drupal/entity_browser": "^2.8",
+        "drupal/entity_browser": "^2.9",
```
Comment (bryanpizzillo) [inline composer.json:45]: lump this into the commit with embed and entity embed and add a note to that issue.

File: composer.json [hunk 2]
```diff
@@ -64,8 +64,8 @@
         "drupal/seckit": "^2.0.1",
         "drupal/shield": "^1.7.0",
         "drupal/simple_sitemap": "^4.1",
-        "drupal/token": "^1.12",
-        "drupal/token_filter": "^2.0",
+        "drupal/token": "^1.13",
```
Comment (bryanpizzillo) [inline composer.json:67]: Add token to the commit with token_filter and add a note to the issue.

File: composer.json [hunk 3]
```diff
@@ -54,7 +54,7 @@
         "drupal/memcache": "^2.5.0",
         "drupal/metatag": "^1.26",
         "drupal/moderation_sidebar": "^1.4",
-        "drupal/paragraphs": "^1.15",
+        "drupal/paragraphs": "^1.16",
```
Comment (bryanpizzillo) [inline composer.json:57]: Add "(#4134) Multiple warning messages when having untranslatable fields" to the claro update commit.

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.install [hunk 4]
```diff
@@ -145,41 +145,48 @@ function _cgov_core_install_permissions(CgovCoreTools $siteHelper) {
...
-      'label' => $block_label,
-      'label_display' => '0',
-    ],
-  ];
-
-  $block = Block::create($data);
-  $block->save();
-
+  foreach ($theme_list as $theme) {
+    if ($theme != 'ncids_trans' && $theme != 'cgov_common') {
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.install:151]: Please add a comment here why we are doing this. Basically saying "Skip doing this if a Drupal core theme is being installed."

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.install [hunk 5]
```diff
@@ -145,41 +145,48 @@ function _cgov_core_install_permissions(CgovCoreTools $siteHelper) {
  *
  * @see system_themes_installed()
  */
-function cgov_core_themes_installed() {
+function cgov_core_themes_installed($theme_list) {
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.install:148]: A separate issue was created, #4172. Just update this commit to:
1. come first in the commit list
2. update the commit message to reference the new number.

**Comment by bryanpizzillo** [comment]
One more thing, Claro update should be the last commit.

**Comment by KateMashkinaNIH** [comment]
This looks good on ODE656

**Review by bryanpizzillo — APPROVED** [review]

**Pull request merged**